grblas/base.py

```diff
@@ -160,7 +161,9 @@ def _update(self, delayed, mask=NULL, accum=NULL, replace=False):
         elif isinstance(accum, ops.BinaryOp):
             accum = accum[self.dtype]
         elif type(accum) is ffi.CData and ops.find_opclass(accum) != ops.UNKNOWN_OPCLASS:
-            pass
+            opclass = ops.find_opclass(accum)
+            if opclass != ops.BinaryOp.__name__:
+                raise TypeError(f'accum must be a BinaryOp, not {opclass}')
```

Comment:
Why can't `accum` be a monoid?

Reply:
That's how the spec is written. Accumulator is always a BinaryOp -- Monoids are not allowed.

I supposed we could get fancy and extract the binaryop out of the monoid for UDFs, but for built-in monoids, that would be harder. I think it's fine to require binaryop. The error message tells the user how to fix the issue.

Reply:
Oh okay.  Sticking to the spec is good.  I know the spec is sometimes flexible in what is allowed.